source/mqtt/Mqtt5ClientCore.cpp

```diff
@@ -642,6 +609,43 @@ namespace Aws
                 }
             }
 
+            Mqtt5to3AdapterOptions::Mqtt5to3AdapterOptions() {}
+
+            ScopedResource<Mqtt5to3AdapterOptions> Mqtt5to3AdapterOptions::NewMqtt5to3AdapterOptions(
+                const Mqtt5ClientOptions &options) noexcept
+            {
+                Allocator *allocator = options.m_allocator;
+                ScopedResource<Mqtt5to3AdapterOptions> adapterOptions = ScopedResource<Mqtt5to3AdapterOptions>(
+                    Crt::New<Mqtt5to3AdapterOptions>(allocator),
+                    [allocator](Mqtt5to3AdapterOptions *options) { Crt::Delete(options, allocator); });
+                adapterOptions->m_mqtt3options.allocator = options.m_allocator;
+                adapterOptions->m_mqtt3options.hostName = options.m_hostName.c_str();
+                adapterOptions->m_mqtt3options.port = options.m_port;
+                adapterOptions->m_mqtt3options.socketOptions = options.m_socketOptions;
+                if (options.m_proxyOptions.has_value())
+                    adapterOptions->m_proxyOptions = options.m_proxyOptions.value();
+                if (options.m_tlsConnectionOptions.has_value())
+                {
+                    adapterOptions->m_mqtt3options.tlsConnectionOptions = options.m_tlsConnectionOptions.value();
+                    adapterOptions->m_mqtt3options.useTls = true;
+                }
+                if (options.websocketHandshakeTransform)
+                {
+                    adapterOptions->m_mqtt3options.useWebsocket = true;
+
+                    auto signerTransform = [&options](
+                                               std::shared_ptr<Crt::Http::HttpRequest> req,
+                                               const Crt::Mqtt::OnWebSocketHandshakeInterceptComplete &onComplete) {
+                        options.websocketHandshakeTransform(std::move(req), onComplete);
+                    };
```

Comment:
Same as above: we use `options` by its address, but the `Mqtt5Client` class description doesn't indicate that users have to keep `options` alive after the client creation.